### DISCUSSION TITLE
My config is not loaded when using asynchronous

### DISCUSSION THREAD

**Discussion by CristoAMH** [wbthomason/packer.nvim#438] [Q&A]
I am trying to use Telescope just when it is invoked the command.
```lua
use {
      'nvim-telescope/telescope.nvim',
      cmd = 'Telescope',
      requires = {'nvim-lua/popup.nvim', 'nvim-lua/plenary.nvim'},
      config = require('config.telescope')
  }
```
This is the content inside config.telescope
```lua
local utils = require('utils')
local telescope = require("telescope")

telescope.setup({})

utils.map('n', '<leader>ff', "<cmd>Telescope find_files<cr>")
utils.map('n', '<leader>fg', "<cmd>Telescope live_grep<cr>")
utils.map('n', '<leader>fb', "<cmd>Telescope buffers<cr>")
utils.map('n', '<leader>fh', "<cmd>Telescope help_tags<cr>")
utils.map('n', '<leader>go', "<cmd>Telescope git_status<cr>")
utils.map('n', '<leader>gb', "<cmd>Telescope git_branches<cr>")
utils.map('n', '<leader>gc', "<cmd>Telescope git_commits<cr>")
utils.map('n', '<leader>gC', "<cmd>Telescope git_bcommits<cr>")

```
but it doesn't work and I have no error, any ideas?

**Comment by nanotee** [comment]
Try
```lua
config = "require('config.telescope')"
```
or
```lua
config = function()
	require('config.telescope')
end
```
Right now you're feeding the return value of `require()` to the `config` key, which is why it doesn't work

**Reply by CristoAMH** [reply]
@nanotee I have tried both options but none of them works and also I tried [[require('config.telescope')]].
If it helps I have this branch with my dotfiles. https://github.com/CristoAMH/DDots/blob/packer/.config/nvim/lua/plugins.lua#L26.

**Comment by nanotee** [comment]
Ah, right, since you lazy-load with a command, your config doesn't get sourced until you run `:Telescope`.
I think adding `keys = {'<Leader>'}` should do the trick

**Comment by wbthomason** [comment]
@nanotee's answer is correct - the `config` key gets run **after** the plugin is loaded, which only happens when `:Telescope` is run in your current specification.

What I would suggest is actually to split your `telescope` config into two parts, one which makes the keymaps and one which calls `telescope.setup`. Source the first part as a `setup` key in your specification, and the second part as a `config` key. That should give you the lazy-loading and configuration you desire.

**Reply by CristoAMH** [reply]
@nanotee @wbthomason  you both are totally right and it have so much sense. I did split my conf and it worked like a charm.
